binding.gyp

```diff
@@ -97,10 +96,56 @@
       'src/utilities.cc',
       'src/sharp.cc'
     ],
-    'include_dirs': [
-      '<!(node -p "require(\'node-addon-api\').include_dir")',
-    ],
     'conditions': [
+      ['OS == "emscripten"', {
+        # .js because that's how Emscripten knows what to build, and .node
+        # in front so that `require('[...].node')` would just work.
+        'product_extension': 'node.js',
+        'defines': [
+          # Limit to 1 async task to avoid stealing threads from the pool
+          # that are intended for libvips.
+          'EMNAPI_WORKER_POOL_SIZE=1',
+        ],
+        'cflags': ['-g2'],
+        'ldflags': [
+          '-g2',
+          '-fexceptions',
+          # We don't know in advance how large images will be, best to allow growth
+          # rather than overallocate in resource-constrained environments.
+          '-sALLOW_MEMORY_GROWTH',
+          # Building for Node.js, we want sync instantiation for compat with native
+          '-sWASM_ASYNC_COMPILATION=0',
+          # Re-export emnapi bindings as the main exports
+          '--pre-js=<!(node -p "require.resolve(\'./wasm-scripts/pre.js\')")',
+          # Using JS implementation of text decoder is actually faster when using pthreads.
+          '-sTEXTDECODER=0',
+          # Support 64-bit integers.
+          '-sWASM_BIGINT',
+          # CPU cores + 3 extra threads for libvips + 1 extra thread for an emnapi async task.
+          '-sPTHREAD_POOL_SIZE="vipsConcurrency+3+1"',
+          '-sPTHREAD_POOL_SIZE_STRICT=2',
+          # Don't wait for the pthread pool to keep initialization synchronous.
+          '-sPTHREAD_POOL_DELAY_LOAD',
+
+          # We're building only for Node.js for now
+          '-sENVIRONMENT=node',
+          # Propagate filesystem to Node.js.
+          '-sNODERAWFS',
+          # Exit helper
+          '-sEXPORTED_FUNCTIONS=["_vips_shutdown"]',
+        ],
+        'link_settings': {
+          # Not sure why, but AOM is not linked implicitly into Sharp.
+          'libraries': ['<!@(PKG_CONFIG_PATH="<(pkg_config_path)" pkg-config --libs aom)']
+        },
```

Comment:
aom is a private dependency of libheif, so this can probably be fixed by passing the `--static` flag to pkg-config. For example:
```console
$ cd wasm-vips/
$ export PKG_CONFIG_PATH="build/target/lib/pkgconfig"
$ pkg-config --libs vips-cpp | grep -c aom
0
$ pkg-config --static --libs vips-cpp | grep -c aom
1
```

The `runtime_link` in `binding.gyp` controls this behavior when  `use_global_libvips == true` (I'm not sure if this is the case for Wasm).
https://github.com/lovell/sharp/blob/82c0159937666918a0330ecb0f3568de10517024/binding.gyp#L153-L157
https://github.com/lovell/sharp/blob/82c0159937666918a0330ecb0f3568de10517024/binding.gyp#L79

Reply:
Interesting! Let me try that.